config/confighttp/confighttp.go

```diff
@@ -62,14 +62,19 @@ type HTTPClientSettings struct {
 
 // ToClient creates an HTTP client.
 func (hcs *HTTPClientSettings) ToClient(ext map[config.ComponentID]component.Extension) (*http.Client, error) {
-	tlsCfg, err := hcs.TLSSetting.LoadTLSConfig()
-	if err != nil {
-		return nil, err
-	}
+	var err error
 	transport := http.DefaultTransport.(*http.Transport).Clone()
-	if tlsCfg != nil {
-		transport.TLSClientConfig = tlsCfg
+
+	if hcs.TLSSetting != nil {
```

Comment:
There seems to be some inconsistency between how gRPC and HTTP handle nil `tls` section. gRPC looks at the scheme to decide whether to enable tls, HTTP does not do that.
Would be good to fix in a separate PR to make sure gRPC and HTTP clients behave the same way.